src/adagrams.js

```diff
@@ -1,15 +1,139 @@
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
+const LETTER_SCORE = {
+  AEIOUNRST: 1,
+  DG: 2,
+  BCMP: 3,
+  FHVWY: 4,
+  K: 5,
+  JX: 8,
+  QZ: 10,
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  // # not sure why not be able to use deep copy.
+  // const copy_pool = structuredClone(LETTER_POOL);
+  const copyPool = { ...LETTER_POOL };
+  let randomTenLetters = [];
+  const keysArr = Object.keys(copyPool);
+
+  let randomKey;
+  while (randomTenLetters.length < 10) {
+    randomKey = keysArr[Math.floor(Math.random() * keysArr.length)];
+    if (copyPool[randomKey] > 0) {
+      randomTenLetters.push(randomKey);
+      copyPool[randomKey] -= 1;
+    }
+  }
+  return randomTenLetters;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const inputArr = input.split("");
+  if (inputArr.length > lettersInHand.length) {
+    return false;
+  }
+
+  for (let i = 0; i < inputArr.length; ++i) {
+    if (lettersInHand.includes(inputArr[i]) === false) {
+      return false;
+    }
+    let index = lettersInHand.indexOf(inputArr[i]);
+    lettersInHand.splice(index, index + 1);
+  }
+  return true;
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+  // if word is empty str return 0
+  if (word === "") return 0;
+
+  word = word.toUpperCase();
+  let total = 0;
+  let keysArr = Object.keys(LETTER_SCORE);
+
+  for (let i = 0; i < word.length; ++i) {
```

Comment:
If we won't be reassigning the loop variable in the body of a loop, I recommend using `const` over `let` to declare the variable.